### ISSUE TITLE
heavy bot improvements

### ISSUE THREAD

**Issue by iraizo** [nullworks/cathook#1398]
**Quick Description**

Heavy bots and bots in general look very suspicious when pushing the cart because of its little movements they make, what i would suggest is that for heavies they rev up for slower speed (and because you should be reved up on the payload in the first place). My second suggestion would be that when the aimbot locks onto somebody there would be a mode that moves the crosshair to the target before it starts shooting/ silent aiming, based on the distance between the crosshair and the player, im pretty sure this would solve the issue that when a bot locks onto a target, and then gets either killed or moves out of sight, the bot snaps directly into the direction of the navbot, instead it should rather turn slowly because sometimes it just tries to move into the opposite direction and just does a 180 which looks suspicious.

**The Issue**

- Suspicious movement
- Suspicious aim

**My Suggestion for Developers**

There should be a slider indicating how fast the bot will move the crosshair to the player.

**Additional Context**

None really

**Comment by EngineerGaming69** [comment]
im pretty sure all of this is being fixed in the legit-bot update.

**Comment by EngineerGaming69** [comment]
also "There should be a slider indicating how fast the bot will move the crosshair to the player." is a thing. slow aim in aimbot and i always have it at 2 and when spectated its at 4

**Comment by iraizo** [comment]
> also "There should be a slider indicating how fast the bot will move the crosshair to the player." is a thing. slow aim in aimbot and i always have it at 2 and when spectated its at 4

This barely works, I also meant that it should be relative to the actual player distance kind of like a "multiplier".

**Comment by EngineerGaming69** [comment]
> > also "There should be a slider indicating how fast the bot will move the crosshair to the player." is a thing. slow aim in aimbot and i always have it at 2 and when spectated its at 4
> 
> This barely works, I also meant that it should be relative to the actual player distance kind of like a "multiplier".

if it was like a multiplier it would look very robotic. being very slow and very fast like a magnet. The slow aim works fine for me

**Comment by iraizo** [comment]
> if it was like a multiplier it would look very robotic. being very slow and very fast like a magnet. The slow aim works fine for me
Well thats pretty much how a human moves, the closer you are the slower you aim to have accuracy, also slow aim just misses or is well.. too slow to even target somebody or just doesent shoot at all at times, im pretty sure there has been an issue with that already

**Comment by BenCat07** [comment]
slow aim has a mathematical flaw, for value 2 it cuts the distance in half every tick, meaning it will actually never reach the center, the higher the number the worse the effect, also no overshoot

**Comment by Auditormadness9** [comment]
Doesn't slow aim already do this?

**Comment by iraizo** [comment]
> Doesn't slow aim already do this?

Please read my comment and post.

**Comment by Auditormadness9** [comment]
> > Doesn't slow aim already do this?
> 
> Please read my comment and post.

I did but I'm still not best at understanding it since I never legit with bots. I'm assuming you mean after the finish aiming they snap back to their "Look at Path"?

**Comment by iraizo** [comment]
> I did but I'm still not best at understanding it since I never legit with bots. I'm assuming you mean after they finish aiming they snap back to their "Look at Path"?

Yes lets imagine a bot is pushing the card on an empty map that just goes foward, the enemy is behind the bot and is snapping directly onto the target (of course with slow aimbot this isnt a problem but slow aimbot is still inaccurate as hell and is basically unusable). Now after he got aimbotted it will directly snap back into where the bot is trying to move which is probaly 180 Degrees from the enemy (because the enemy is behind him and he tries to push the card forwards) which looks very suspicous.

**Comment by Auditormadness9** [comment]
> Yes lets imagine a bot is pushing the card on an empty map that just goes foward, the enemy is behind the bot and is snapping directly onto the target (of course with slow aimbot this isnt a problem but slow aimbot is still inaccurate as hell and is basically unusable). Now after he got aimbotted it will directly snap back into where the bot is trying to move which is probaly 180 Degrees from the enemy (because the enemy is behind him and he tries to push the card forwards) which looks very suspicous.

Okay so "slow look at path" is what we need

**Comment by iraizo** [comment]
> Okay so "slow look at path" is what we need
Yes and a proper slow aimbot fix
